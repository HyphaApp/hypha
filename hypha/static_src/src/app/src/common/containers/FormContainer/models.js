import * as Immutable from 'seamless-immutable';

export const formInitialState = Immutable.from({
    values: {},
    errors: {},
    constraints: {},
    readyToSubmit: false
});

const initialState = Immutable.from({
    forms: {}
});

export default initialState;
