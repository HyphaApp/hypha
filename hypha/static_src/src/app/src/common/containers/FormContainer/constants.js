
export const INITIALIZE_FORM = 'views/Settings/components/FormContainer/INITIALIZE_FORM';
export const UPDATE_FIELD_VALUE = 'views/Settings/components/FormContainer/UPDATE_FIELD_VALUE';
export const ADD_VALIDATION_ERROR = 'views/Settings/components/FormContainer/ADD_VALIDATION_ERROR';


export const VALIDATE_AND_SUBMIT_FORM = 'views/Settings/components/FormContainer/VALIDATE_AND_SUBMIT_FORM';
export const CLEAR_VALIDATION_ERRORS = 'views/Settings/components/FormContainer/CLEAR_VALIDATION_ERRORS';

export const DESTROY_FORM = 'views/Settings/components/FormContainer/DESTROY_FORM';


