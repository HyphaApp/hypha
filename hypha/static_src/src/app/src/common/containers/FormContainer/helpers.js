import {formInitialState} from './models';

export const initializer = (fields, initialValues = false) => {
    let formState = formInitialState
    for(const field of fields) {
        formState = formState.merge(
            {
                values: {
                  [field.kwargs.label]: field.kwargs.initial !== undefined ? field.kwargs.initial : ""
                },
                constraints: {
                  [field.kwargs.label]: getConstraints(field) 
                }
            },
            {deep: true}
        )
    }

      // Add default values for dropdown
  for (const field of fields) {
    if(field.type === "ScoredAnswerField"){
      formState = formState.merge(
        {
          values: {
            [field.kwargs.label]: [field.kwargs.initial[0], field.kwargs.fields[1].choices[0][0]]
          }
        }
      )
    }
    if (field.type === "ChoiceField" && !formState.values[field.kwargs.label]) {
      formState = formState.merge(
        {
          values: {
            [field.kwargs.label]: field.kwargs.choices[0][0]
          },
        },
        { deep: true }
      )
    }
  }  

  if (initialValues) {
      formState = formState.merge({values: {}},{ deep: true });
      for (const fieldId in initialValues) {
        const field = fields.find(field => field.id === fieldId);
        if(field) {
          formState = formState.merge({ values: {
            [field.kwargs.label]: initialValues[fieldId]
          } }, { deep: true });
        }
      }
    }
    

    return formState;
};


const getConstraints = field => {
  let constraints = {}
  if (field.kwargs && field.kwargs.required) {
    constraints["presence"] = { allowEmpty: false }
  }

  if (field.kwargs && field.kwargs.max_length) {
    constraints["length"] = { maximum: true }
  }

  if (field.type === "EmailField") {
    constraints["email"] = true
  }

  if (field.type === "URLField") {
    constraints["url"] = true
  }

  if (field.kwargs && field.kwargs.min_length) {
    if (!constraints["length"]) {
      constraints["length"] = { minimum: true }
    } else {
      constraints["length"]["minimum"] =  true 
    }
  }

  if(field.type === "ScoredAnswerField"){
    constraints = {"scoredAnswerValidator" : true}
  }
  return constraints
} 


export const parseValues = (fields, values) => {
  const formattedValues = {}
  for (const field of fields) {
    formattedValues[field.id] = values[field.kwargs.label]
  }
  return formattedValues
}