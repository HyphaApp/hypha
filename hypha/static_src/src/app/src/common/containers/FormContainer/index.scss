.footer {
    display: flex;
    justify-content: flex-end;
    margin: 20px 5px;
    dnx-button {
        margin-right: 25px;
    }
}

.fieldsContiner {
    display: flex;
    width: 100%;
    flex-wrap: wrap;
    margin: 10px;
    justify-content: space-between;
    
}