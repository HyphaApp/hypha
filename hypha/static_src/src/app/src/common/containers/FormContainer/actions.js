import * as ActionTypes from './constants';

export const initializeFormAction = (formId, form) => ({
    type: ActionTypes.INITIALIZE_FORM,
    formId,
    form
});

export const updateFieldValueAction = (formId, fieldName, fieldValue) => ({
    type: ActionTypes.UPDATE_FIELD_VALUE,
    formId,
    fieldName,
    fieldValue
});

export const addValidationErrorAction = (formId, fieldName, errorMessage) => ({
    type: ActionTypes.ADD_VALIDATION_ERROR,
    formId,
    fieldName,
    errorMessage
});


export const clearValidationErrorAction = formId => ({
    type: ActionTypes.CLEAR_VALIDATION_ERRORS,
    formId
});


export const destroyFormAction = formId => ({
    type: ActionTypes.DESTROY_FORM,
    formId
});


export const validateAndSubmitFormAction = formId => ({
    type: ActionTypes.VALIDATE_AND_SUBMIT_FORM,
    formId
});
