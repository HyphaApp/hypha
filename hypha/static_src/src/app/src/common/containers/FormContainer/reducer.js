import initialState, {formInitialState} from './models';
import * as ActionTypes from './constants';


const FormReducer = (
    state = formInitialState,
    action
) => {
    switch(action.type) {
        case ActionTypes.UPDATE_FIELD_VALUE:
            return state
                .set('readyToSubmit', false)
                .setIn(
                    ['values', action.fieldName], 
                    action.fieldValue);
        case ActionTypes.ADD_VALIDATION_ERROR:
            if (action.errorMessage){
                return state.setIn(
                    ['errors', action.fieldName],
                    action.errorMessage);
            }
            else {
                return state.set('errors', state.errors.without(action.fieldName))
            }
        case ActionTypes.CLEAR_VALIDATION_ERRORS:
            return state.set('errors', {});
        case ActionTypes.VALIDATE_AND_SUBMIT_FORM:
            return state.set('readyToSubmit', true);
        default:
            return state;
    }
}

const FormContainerReducer = (
    state = initialState,
    action
) => {
    switch (action.type) {
        case ActionTypes.INITIALIZE_FORM:
            return state
                .setIn(['forms', action.formId], action.form);
        case ActionTypes.UPDATE_FIELD_VALUE:
        case ActionTypes.VALIDATE_AND_SUBMIT_FORM:
        case ActionTypes.CLEAR_VALIDATION_ERRORS:
        case ActionTypes.ADD_VALIDATION_ERROR:
            return state
                .setIn(
                    ['forms', action.formId],
                    FormReducer(state.forms[action.formId], action)
                );
        default:
            return state;
    }
}


export default FormContainerReducer;
