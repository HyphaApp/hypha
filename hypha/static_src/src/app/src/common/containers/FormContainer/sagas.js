import { takeEvery, put, select } from 'redux-saga/effects';
// eslint-disable-next-line no-unused-vars
import regeneratorRuntime from "regenerator-runtime";

import * as ActionTypes from "./constants";
import * as Actions from "./actions";
import * as Selectors from "./selectors";
import Validator from "./Validator";

export function* validateFields(action){
    const formsInfo = yield select(Selectors.selectFormsInfo);
    const formId = action.formId;
    const validator = new Validator(formsInfo[formId].values, formsInfo[formId].constraints);
    const errors = validator.validate();

    yield put(Actions.clearValidationErrorAction(formId));
    if(errors) {
        for(const field_name in errors){
            yield put(Actions.addValidationErrorAction(formId, field_name, errors[field_name][0] ))
        }
    }
}


export function* validateField(action) {
    const formsInfo = yield select(Selectors.selectFormsInfo);
    const formId = action.formId;
    const validator = new Validator(formsInfo[formId].values, formsInfo[formId].constraints);
    const error = validator.validateSingle(action.fieldName);
    yield put(Actions.addValidationErrorAction(
        formId, 
        action.fieldName,
        !error ? error : error[action.fieldName][0])
    )
}


export default function* formContainerSaga() {
  yield takeEvery(
    ActionTypes.UPDATE_FIELD_VALUE,
    validateField
  );
  yield takeEvery(
    ActionTypes.VALIDATE_AND_SUBMIT_FORM,
    validateFields
  );
}
