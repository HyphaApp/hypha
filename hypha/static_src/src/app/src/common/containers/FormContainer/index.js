
import * as React from "react";
import FormField from "./components/FormField";
import _ from "underscore";
import * as Actions from './actions';
import * as Selectors from './selectors';
import {bindActionCreators, compose} from "redux";
import {connect} from "react-redux";
import {initializer} from "./helpers";
import injectReducer from '@utils/injectReducer'
import injectSaga from '@utils/injectSaga'
import reducer from './reducer';
import saga from './sagas';
import { withRouter } from 'react-router-dom';
import { parseValues} from './helpers';
import PropTypes from 'prop-types';


class FormContainer extends React.Component {
    constructor() {
        super();
        this.onChange = this.onChange.bind(this);
        this.submitFn = null;
    }

    onSubmit = (actionType, submitFn) => () => {
        const { values, errors} = this.formInfo;
      if (_.isEmpty(errors) && actionType === "primary") {
        this.submitFn = submitFn;
        this.props.validateAndSubmitForm(this.props.formId)
      }
      else if (actionType !== "primary"){
        submitFn(parseValues(this.props.metadata.fields, values))
      }
    }
    
    componentDidUpdate() {
        const { values, errors, readyToSubmit} = this.formInfo;
        if(readyToSubmit && _.isEmpty(errors)) {
          this.submitFn(parseValues(this.props.metadata.fields, values))
        }
    }

    componentDidMount() {
        this.props.initialize(
            this.props.formId,
          initializer(this.props.metadata.fields, this.props.metadata.initialValues)
        );
    }

    onChange(name, value)  {
        this.props.updateFieldValue(this.props.formId, name, value)
    }

    get formInfo() {
        return {
            values: this.props.formsInfo[this.props.formId] ? this.props.formsInfo[this.props.formId].values : {},
            errors: this.props.formsInfo[this.props.formId] ? this.props.formsInfo[this.props.formId].errors : {},
            readyToSubmit: this.props.formsInfo[this.props.formId] ? this.props.formsInfo[this.props.formId].readyToSubmit : false
        }
    }

    render() {
        const formFields = this.props.metadata.fields;
        const actions = this.props.metadata.actions;
        const {errors, values} = this.formInfo;
        return <div>
            {this.props.metadata.title && <h3> {this.props.metadata.title} </h3>}
            <div style={{ width: `${this.props.metadata.width}px` }}>
            <form className="form form--with-p-tags form--scoreable">
                {(formFields || []).map(field => {
                    return <FormField 
                      type={field.type}
                      key={field.id}
                      name={field.kwargs.label}
                      kwargs={field.kwargs}
                      onChange={this.onChange}
                      value={values[field.kwargs.label]}
                      error={errors[field.kwargs.label]}
                      widget={field.widget}
                    />
                })}
                </form>
            </div>

            <div > 
                {(actions || [] ).map(action => {
                  let disabled = false;
                  if (action.type === "primary") {
                    disabled = Object.keys(errors).length > 0
                  }
                  return <input
                    className={`button button--submit button--top-space button--${action.type === 'primary' ? 'primary' : 'white'} ${action.type === 'warning' ? 'button--warning' : ''}`}
                    type="button"
                    disabled={disabled}
                    key={action.text}
                    value={action.text}
                    onClick={this.onSubmit(action.type, action.callback)}
                  />
                })}
            </div>
        </div>;
    }
}

FormContainer.propTypes = {
  formsInfo: PropTypes.object,
  initialize: PropTypes.func,
  updateFieldValue: PropTypes.func,
  validateAndSubmitForm: PropTypes.func,
  formId: PropTypes.string,
  metadata: PropTypes.object,
   
}


const mapStateToProps = state => ({
    formsInfo: Selectors.selectFormsInfo(state),
    activeForm: Selectors.selectActiveForm(state)
});

const mapDispatchToProps = dispatch => {
    return bindActionCreators({
        initialize: Actions.initializeFormAction,
        updateFieldValue: Actions.updateFieldValueAction,
        validateAndSubmitForm: Actions.validateAndSubmitFormAction
    }, dispatch);
}


const withConnect = connect(mapStateToProps, mapDispatchToProps)

const withReducer = injectReducer({ key: 'FormContainer', reducer });
const withSaga = injectSaga({ key: 'FormContainer', saga });

export default compose(
  withSaga,
  withReducer,
  withConnect,
  withRouter,
)(FormContainer);