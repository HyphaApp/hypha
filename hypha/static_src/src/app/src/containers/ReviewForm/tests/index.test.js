import React from "react";
import {mount} from "enzyme";
import {Provider} from "react-redux";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import initialState from '../models'
import  {ReviewFormContainer}  from '../index.js';
import LoadingPanel from '@components/LoadingPanel'
enzyme.configure({adapter: new Adapter()});
import FormContainer from '@common/containers/FormContainer';


describe("Test Review form Container", () => {
    it("Should render review form with loading true", () => {
        const initializeAction = jest.fn()
        const wrapper = mount(
                <ReviewFormContainer 
                formData={{loading : true}} 
                initializeAction={initializeAction}/>
        );
        expect(wrapper.find('.container').length).toEqual(1)
        expect(wrapper.find('h3').text()).toBe("Create Review")
        expect(initializeAction).toHaveBeenCalled()
        expect(wrapper.containsMatchingElement(<LoadingPanel />)).toBe(true)
        expect(wrapper).toMatchSnapshot();
    });

    it("Should render review with loading false", () => {
        const initializeAction = jest.fn()
        const toggleReviewForm =jest.fn()
        const deleteReview = jest.fn()
        const submitReview = jest.fn()
        const updateReview = jest.fn()
        const wrapper = shallow(
                <ReviewFormContainer 
                formData={{
                    loading : false,
                    saveAsDraft : true,
                    initialValues : [],
                    metaStructure : []}} 
                initializeAction={initializeAction} 
                submissionID={1} 
                reviewId={2}
                toggleReviewForm={toggleReviewForm}
                deleteReview={deleteReview}
                submitReview={submitReview}
                updateReview={updateReview}
                />
        );
        expect(wrapper.find('.container').length).toEqual(1)
        expect(wrapper.find('h3').text()).toBe("Update Review")
        expect(initializeAction).toHaveBeenCalled()
        expect(initializeAction).toHaveBeenCalledWith(1,2)
        expect(toggleReviewForm).not.toHaveBeenCalled()
        expect(deleteReview).not.toHaveBeenCalled()
        expect(submitReview).not.toHaveBeenCalled()
        expect(updateReview).not.toHaveBeenCalled()
        expect(wrapper.containsMatchingElement(<FormContainer />)).toBe(true)
        expect(wrapper).toMatchSnapshot();
    });

    it("Should render review with loading false & saveAsDraft false", () => {
        const initializeAction = jest.fn()
        const toggleReviewForm =jest.fn()
        const deleteReview = jest.fn()
        const submitReview = jest.fn()
        const updateReview = jest.fn()
        const wrapper = shallow(
                <ReviewFormContainer 
                formData={{
                    loading : false,
                    saveAsDraft : false,
                    initialValues : [],
                    metaStructure : [{type : "type1"}, {type : "TypedChoiceField"}]}} 
                initializeAction={initializeAction} 
                submissionID={1} 
                currentDetermination={3}
                toggleReviewForm={toggleReviewForm}
                deleteReview={deleteReview}
                submitReview={submitReview}
                updateReview={updateReview}
                />
        );
        expect(wrapper.find('.container').length).toEqual(1)
        expect(wrapper.find('h3').text()).toBe("Create Review")
        expect(initializeAction).toHaveBeenCalled()
        expect(initializeAction).toHaveBeenCalledWith(1,null)
        expect(wrapper.containsMatchingElement(<FormContainer />)).toBe(true)
        expect(wrapper).toMatchSnapshot();
    });

    it("Should render review with loading false & saveAsDraft false & review id exists", () => {
        const initializeAction = jest.fn()
        const toggleReviewForm =jest.fn()
        const deleteReview = jest.fn()
        const submitReview = jest.fn()
        const updateReview = jest.fn()
        const wrapper = shallow(
                <ReviewFormContainer 
                formData={{
                    loading : false,
                    saveAsDraft : false,
                    initialValues : [],
                    metaStructure : []}} 
                initializeAction={initializeAction} 
                submissionID={1} 
                reviewId={2}
                toggleReviewForm={toggleReviewForm}
                deleteReview={deleteReview}
                submitReview={submitReview}
                updateReview={updateReview}
                />
        );
        expect(wrapper.find('.container').length).toEqual(1)
        expect(wrapper.find('h3').text()).toBe("Update Review")
        expect(initializeAction).toHaveBeenCalled()
        expect(initializeAction).toHaveBeenCalledWith(1,2)
        expect(toggleReviewForm).not.toHaveBeenCalled()
        expect(deleteReview).not.toHaveBeenCalled()
        expect(submitReview).not.toHaveBeenCalled()
        expect(updateReview).not.toHaveBeenCalled()
        expect(wrapper.containsMatchingElement(<FormContainer />)).toBe(true)
        expect(wrapper).toMatchSnapshot();
    });
});
