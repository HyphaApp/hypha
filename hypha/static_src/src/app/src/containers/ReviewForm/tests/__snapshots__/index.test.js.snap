// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Review form Container Should render review form with loading true 1`] = `
<ReviewFormContainer
  formData={
    Object {
      "loading": true,
    }
  }
  initializeAction={
    [MockFunction] {
      "calls": Array [
        Array [
          undefined,
          null,
        ],
      ],
    }
  }
>
  <div
    className="container"
  >
    <h3>
      Create Review
    </h3>
    <LoadingPanel>
      <div
        className="loading-panel"
      >
        <div
          className="loading-panel__text"
        >
          <h5>
            Loading...
          </h5>
        </div>
        <div
          className="loading-panel__icon"
        >
          <OTFLoadingIcon>
            <svg
              className="logo-loading"
              viewBox="0 0 45 40"
            >
              <g
                fill="none"
                fillRule="nonzero"
              >
                <rect
                  className="logo-part logo-part-1"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(10 0)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-2"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(5 5)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-3"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(0 10)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-4"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(0 15)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-5"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(0 20)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-6"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(0 25)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-7"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(5 30)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-8"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(10 35)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-9"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(30 0)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-10"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(35 5)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-11"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(40 10)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-12"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(40 15)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-13"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(40 20)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-14"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(40 25)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-15"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(35 30)"
                  width="5"
                />
                <rect
                  className="logo-part logo-part-16"
                  fill="#25AAE1"
                  height="5"
                  transform="translate(30 35)"
                  width="5"
                />
              </g>
            </svg>
          </OTFLoadingIcon>
        </div>
      </div>
    </LoadingPanel>
  </div>
</ReviewFormContainer>
`;

exports[`Test Review form Container Should render review with loading false & saveAsDraft false & review id exists 1`] = `
<div
  className="container"
>
  <h3>
    Update Review
  </h3>
  <withSaga(withReducer(Connect(withRouter(FormContainer))))
    formId="myIntialForm"
    metadata={
      Object {
        "actions": Array [
          Object {
            "callback": [Function],
            "text": "back",
            "type": "secondary",
          },
          Object {
            "callback": [Function],
            "text": "Update",
            "type": "primary",
          },
          Object {
            "callback": [Function],
            "text": "Delete",
            "type": "warning",
          },
        ],
        "fields": Array [],
        "initialValues": Array [],
      }
    }
  />
</div>
`;

exports[`Test Review form Container Should render review with loading false & saveAsDraft false 1`] = `
<div
  className="container"
>
  <h3>
    Create Review
  </h3>
  <withSaga(withReducer(Connect(withRouter(FormContainer))))
    formId="myIntialForm"
    metadata={
      Object {
        "actions": Array [
          Object {
            "callback": [Function],
            "text": "back",
            "type": "secondary",
          },
          Object {
            "callback": [Function],
            "text": "Save Draft",
            "type": "secondary",
          },
          Object {
            "callback": [Function],
            "text": "Create",
            "type": "primary",
          },
        ],
        "fields": Array [
          Object {
            "type": "type1",
          },
          Object {
            "type": "TypedChoiceField",
          },
        ],
        "initialValues": Array [],
      }
    }
  />
</div>
`;

exports[`Test Review form Container Should render review with loading false 1`] = `
<div
  className="container"
>
  <h3>
    Update Review
  </h3>
  <withSaga(withReducer(Connect(withRouter(FormContainer))))
    formId="myIntialForm"
    metadata={
      Object {
        "actions": Array [
          Object {
            "callback": [Function],
            "text": "back",
            "type": "secondary",
          },
          Object {
            "callback": [Function],
            "text": "Update Draft",
            "type": "secondary",
          },
          Object {
            "callback": [Function],
            "text": "Update",
            "type": "primary",
          },
          Object {
            "callback": [Function],
            "text": "Delete",
            "type": "warning",
          },
        ],
        "fields": Array [],
        "initialValues": Array [],
      }
    }
  />
</div>
`;
